packages/babel-helpers/src/helpers.js

```diff
@@ -216,8 +216,10 @@ helpers.asyncGeneratorDelegate = template(`
 helpers.asyncToGenerator = template(`
   (function (fn) {
     return function () {
-      var gen = fn.apply(this, arguments);
+      var _args = arguments;
+      var _promiseWrapper = this;
       return new Promise(function (resolve, reject) {
```

Comment:
If you want to change this to an arrow function, that's fine with me. Seems like it would be worth it.